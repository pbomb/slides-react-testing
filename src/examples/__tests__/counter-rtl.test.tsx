import { render, Simulate } from 'react-testing-library';
import * as React from 'react';
import Counter from '../counter';

test('renders enabled buttons by default', () => {
  const { container } = render(<Counter />);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders disabled buttons', () => {
  const { container } = render(<Counter disabled={true} />);
  expect(container.firstChild).toMatchSnapshot();
});

test('increments the counter when clicking the + button', () => {
  const { container, getByTestId } = render(<Counter />);
  Simulate.click(getByTestId('increment'));
  expect(container.firstChild).toMatchSnapshot();
});

test('decrements the counter when clicking the - button', () => {
  const { container, getByTestId } = render(<Counter />);
  Simulate.click(getByTestId('decrement'));
  expect(container.firstChild).toMatchSnapshot();
});
