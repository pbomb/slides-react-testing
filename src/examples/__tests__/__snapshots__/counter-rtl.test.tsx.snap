// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decrements the counter when clicking the - button 1`] = `
.css-0,
[data-css-0] {
  color: red;
}

<div>
  <button
    data-testid="decrement"
  >
    -
  </button>
  <div
    class="css-0"
    data-testid="count"
  >
    -1
  </div>
  <button
    data-testid="increment"
  >
    +
  </button>
</div>
`;

exports[`increments the counter when clicking the + button 1`] = `
.css-0,
[data-css-0] {
  color: green;
}

<div>
  <button
    data-testid="decrement"
  >
    -
  </button>
  <div
    class="css-0"
    data-testid="count"
  >
    1
  </div>
  <button
    data-testid="increment"
  >
    +
  </button>
</div>
`;

exports[`renders disabled buttons 1`] = `
.css-0,
[data-css-0] {
  color: black;
}

<div>
  <button
    data-testid="decrement"
    disabled=""
  >
    -
  </button>
  <div
    class="css-0"
    data-testid="count"
  >
    0
  </div>
  <button
    data-testid="increment"
    disabled=""
  >
    +
  </button>
</div>
`;

exports[`renders enabled buttons by default 1`] = `
.css-0,
[data-css-0] {
  color: black;
}

<div>
  <button
    data-testid="decrement"
  >
    -
  </button>
  <div
    class="css-0"
    data-testid="count"
  >
    0
  </div>
  <button
    data-testid="increment"
  >
    +
  </button>
</div>
`;
